storage/remote/write_handler_test.go

```diff
@@ -312,30 +310,62 @@ func TestRemoteWriteHandler_V2Message(t *testing.T) {
 	resp := recorder.Result()
 	require.Equal(t, http.StatusNoContent, resp.StatusCode)
 
+	b := labels.NewScratchBuilder(0)
 	i := 0
 	j := 0
 	k := 0
-	// the reduced write request is equivalent to the write request fixture.
-	// we can use it for
 	for _, ts := range writeV2RequestFixture.Timeseries {
-		ls := writev2.DesymbolizeLabels(ts.LabelsRefs, writeV2RequestFixture.Symbols)
+		ls := ts.ToLabels(&b, writeV2RequestFixture.Symbols)
 		for _, s := range ts.Samples {
-			require.Equal(t, mockSample{ls, s.Timestamp, s.Value}, appendable.samples[i])
+			// NOTE(bwplotka): In this case I got errors due to correct labels.Labels.String(), but with dedupelabels tag
```

Comment:
Please review rest, but this one is weird. Have to investigate more.

cc @bboreham wonder if you see anything obviously wrong in this PR causing only dedup labels and only in this certain case to kind of work (same labels!!!) but have different internal symbols? Is it a problem?

Reply:
Asked on Slack too https://cloud-native.slack.com/archives/C01AUBA4PFE/p1719401691368089

Reply:
This is what `testutil.RequireEqual` is for. https://github.com/prometheus/prometheus/blob/2dd07fbb1bfe8ebeca8ea11f4623e0f5faed2236/util/testutil/cmp.go#L27

Reply:
Fixed by requireEqual function